docs/rfcs/749-unify-path-behavior.md

```diff
@@ -0,0 +1,88 @@
+- Author: JinnyYi <github.com/JinnyYi>
+- Start Date: 2021-09-08
+- RFC PR: [beyondstorage/go-storage#749](https://github.com/beyondstorage/go-storage/pull/749)
+- Tracking Issue: [beyondstorage/go-storage#0](https://github.com/beyondstorage/go-storage/issues/0)
+
+# GSP-749: Unify Path Behavior
+
+Previous Discussion:
+
+- [Specify the behavior of Path](https://forum.beyondstorage.io/t/topic/195)
+- [OS-specific path separator problem](https://github.com/beyondstorage/go-service-ftp/issues/22)
+
+## Background
+
+Unix and Windows all use a different syntax for filesystem paths. Normally, when specifying file paths on Windows, we would use backslashes(`\`). However, in Java, and many other places outside the Windows world, backslashes are the escape character, so we have to double them up, like `C:\\trash\\blah\\blah`. Forward slashes(`/`), on the other hand, do not need to be doubled up and work on both Windows and Unix. There is no harm in having double forward slashes. They do nothing to the path and just take up space (`//` is equivalent to `/./`).
+
+In our services, we have to handle paths like path splicing and path conversion. And here comes the problems:
+
+- Different path separators may exist in a path at the same time, is it possible or how to use cross-platform paths for users?
+- For fs storage service on Windows platform, the behavior of `work_dir` is undefined. Drive letters in connection string cannot be handled for Windows at this stage.
+
+## Proposal
+
+### Absolute Path and Relative Path
+
+All our services should support two kinds of path:
+
+- Absolute Path:
+  - For Unix, the absolute path starts with `/`.
+  - For Windows, the absolute path starts with a drive letter, `/` is the current drive.
+- Relative Path: 
+  - The relative path is for the working directory `WorkDir`.
+  - Path with prefix `./` or `../` is allowed.
+
+### WorkDir and path
+
+`WorkDir` specifies the working directory of the process.
+
+- For file system, `WorkDir` SHOULD be a passed in absolute path or relative to the current directory of the process.

```

Comment:
Do we need to restrict the `WorkDir` to be an absolute path?  So that for fs in Windows, `WorkDir` should be `c:\a\b`?

Reply:
> Do we need to restrict the `WorkDir` to be an absolute path?

Yes, I think we need to do this.